custom/inddex/food.py

```diff
@@ -334,32 +331,24 @@ def recipe_id(self):
             return self.caseid
         return self.recipe_case_id or 'NO_RECIPE'
 
-    _total_grams = 'NOT_YET_COMPUTED'
-    @property
-    def total_grams(self):
-        # This property must be computed later, as it depends on other rows
-        if self._total_grams == 'NOT_YET_COMPUTED':
-            raise AssertionError("total_grams has not yet been computed")
-        return self._total_grams
-
-    @total_grams.setter
-    def total_grams(self, val):
-        self._total_grams = val
-
     def __getattr__(self, name):
         if name in _INDICATORS_BY_SLUG:
+            indicator = _INDICATORS_BY_SLUG[name]
+            if indicator.is_calculated_later:
+                if not self.enrichment_complete:
+                    raise AttributeError(f"{name} hasn't yet been set. It will be "
+                                        "calculated outside the scope of FoodRow.")
+                return None
```

Comment:
@kaapstorm I forgot about this - not all of the `CALCULATED_LATER` indicators are set in all instances.  I just changed the approach slightly.

Reply:
If `self.enrichment_complete` is False, but foo indicator has been calculated, then when a user asks for row.foo, this code will not be reached.

This will only return None if  `self.enrichment_complete` is True and foo indicator has not been calculated -- which I don't think is what you had in mind.

Reply:
Wait. Maybe it is.

Reply:
I take it back. That is exactly what you had in mind.

Reply:
(Sorry for the stream-of-consciousness comments.)

Reply:
that was a rollercoaster!